purchase_reception_status/models/purchase_order.py

```diff
@@ -0,0 +1,57 @@
+# Copyright 2020 Akretion France (http://www.akretion.com/)
+# @author: Alexis de Lattre <alexis.delattre@akretion.com>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import api, fields, models
+from odoo.tools import float_compare
+
+
+class PurchaseOrder(models.Model):
+    _inherit = "purchase.order"
+
+    reception_status = fields.Selection(
+        [
+            ("no", "Nothing Received"),
+            ("partial", "Partially Received"),
+            ("received", "Fully Received"),
+        ],
+        compute="_compute_reception_status",
+        store=True,
+    )
+    force_received = fields.Boolean(
+        readonly=True,
+        states={"done": [("readonly", False)]},
```

Comment:
Since Odoo 17, property purchase.order.force_received.states is no longer supported.

Reply:
I think it can be removed and XML can updated with readonly="state != 'done'"

Reply:
@Rad0van Can you please address this comment?

Reply:
Done